BOJ/14567_지혁.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+#include <vector>
+#include <queue>
+
+using namespace std;
+
+int N, M;
+vector<int> in_degree, ans;
+vector<int> graph[1001];
+
+void topology_sort() {
+    queue<pair<int, int> > q;
+
+    for(int i = 1; i <= N; i++) {
+        if(in_degree[i] == 0) {
+            q.push(make_pair(i, 1));
+            ans[i] = 1;
+        }
+    }
+
+    for(int i = 1; i <= N; i++) {
+        pair<int, int> cur = q.front();
+        q.pop();
+        for(int j = 0; j < graph[cur.first].size(); j++) {
+            int next = graph[cur.first][j];
+            if(--in_degree[next] == 0) {
+                q.push(make_pair(next, cur.second + 1));
+                ans[next] = cur.second + 1;
+            }
+        }
+    }
```

Comment:
queue를 사용하기 위한 반복문을 for로 작성해 위 코드에서는 정상 작동하지만 다른 경우에는 queue 내 모든 원소를 탐색하지 못하거나 중간에 에러가 발생할 수 있어 보여요. 아래와 같이 수정하면 좋을 거 같아요!
```suggestion
    while(!q.empty()) {
        pair<int, int> cur = q.front();
        q.pop();
        for(int j = 0; j < graph[cur.first].size(); j++) {
            int next = graph[cur.first][j];
            if(--in_degree[next] == 0) {
                q.push(make_pair(next, cur.second + 1));
                ans[next] = cur.second + 1;
            }
        }
    }
```